lib/jarvis/commands/publish.rb

```diff
@@ -29,28 +42,139 @@ def execute
 
     logger.info("Cloning repo", :url => project.git_url)
     git = Jarvis::Git.clone_repo(project.git_url, workdir)
-
+    
     branches.each do |branch|
+
+      unless force?
+        build = build_report(project)
+        workdir_sha1 = Jarvis::Git.sha1(workdir) 
+
+        if build.sha1 == workdir_sha1
+          raise CommitHashDontMatch, "workdir_sha1: #{workdir_sha1}, build_sha1: #{build_sha1}"
+        end
+
+        if !build.success?
+          raise CIBuildFail, "Expecting success got #{build.status}"
+        end
+      end
+
       logger.info("Switching branches", :branch => branch)
       git.checkout(branch)
       context = logger.context
       context[:operation] = "publish"
       context[:branch] = branch
 
-      TASKS.each do |command|
-        context[:command] = command
-        Jarvis.execute(command, logger, git.dir)
+      TASKS.each do |command, condition|
+        if condition.call(workdir)
+          context[:command] = command
+          Jarvis.execute(command, logger, git.dir)
 
-        # Clear the logs if it was successful
-        logs.clear unless logger.debug?
+          # Clear the logs if it was successful
+          logs.clear unless logger.debug?
+        end
       end
       context.clear()
       git.reset
       git.clean(force: true)
+
+      check_version_match
```

Comment:
Well, this need to be done after the `publish_gem` to check if the local version is available on rubygems?